test/mapping.test.js

```diff
@@ -0,0 +1,108 @@
+process.env.NODE_ENV = 'test';
+require('should');
+require('./init');
+
+var async = require('async');
+
+var db;
+
+before(function() {
+  db = getDataSource();
+});
+
+describe('Mapping models', function() {
+  it('should honor the mssql settings for table/column', function(done) {
+
+    var schema =
+    {
+      'name': 'TestInventory',
+      'options': {
+        'idInjection': false,
+        'mssql': {
+          'schema': 'dbo', 'table': 'INVENTORYTEST'
+        }
+      },
+      'properties': {
+        'productId': {
+          'type': 'Number', 'id': true, generated: true,
+          'mssql': {
+            'columnName': 'PRODUCT_ID',
+            'dataType': 'uniqueidentifier',
+            'nullable': 'N'
+          }
+        },
+        'locationId': {
+          'type': 'String', 'required': true, 'length': 20,
+          'mssql': {
+            'columnName': 'LOCATION_ID', 'dataType': 'nvarchar',
+            'nullable': 'N'
+          }
+        },
+        'available': {
+          'type': 'Number', 'required': false,
+          'mssql': {
+            'columnName': 'AVAILABLE', 'dataType': 'int', 'nullable': 'Y'
+          }
+        },
+        'total': {
+          'type': 'Number', 'required': false,
+          'mssql': {
+            'columnName': 'TOTAL', 'dataType': 'int', 'nullable': 'Y'
+          }
+        }
+      }
+    };
+    var models = db.modelBuilder.buildModels(schema);
+    // console.log(models);
+    var Model = models['TestInventory'];
+    Model.attachTo(db);
+
+    db.automigrate(function(err, data) {
+      async.series([
+        function(callback) {
+          Model.destroyAll(callback);
+        },
+        function(callback) {
+          Model.create({locationId: 'l001',
+            available: 10, total: 50}, callback);
+        },
+        function(callback) {
+          Model.create({locationId: 'l002',
+            available: 30, total: 40}, callback);
+        },
+        function(callback) {
+          Model.create({locationId: 'l001',
+            available: 15, total: 30}, callback);
```

Comment:
I find this style confusing. IMO you should either put all key/value pairs on a single line, or else put each key/value pair on a new line.

```js
Model.create(
  { locationId: 'l001', available: 15, total: 30 },
  callback);
```

The same comment applies to L66-L67 and L70-L71.